packages/lit-dev-content/samples/tutorials/first-component/07/before/word-viewer.ts

```diff
@@ -0,0 +1,52 @@
+import { css, html, LitElement } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+
+@customElement('word-viewer')
+class WordViewer extends LitElement {
+  static styles = css`
+    :host {
+      background-color: white;
+      color: violet;
+      cursor: pointer;
+      display: block;
+    }
+    pre {
+      padding: 0.2em;
+    }
+    /* TODO: Add .backwards styles */
+  `;
+
+  @state() private playDirection = 1;
+  @state() private idx = 0;
+  @property() words: string = 'initial value';
+
+  private intervalTimer?: number;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.intervalTimer = setInterval(this.tickToNextWord, 1000);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.intervalTimer);
+    this.intervalTimer = undefined;
+  }
+
+  render() {
+    const splitWords = this.words.split('.');
+    const word = splitWords[((this.idx % splitWords.length) + splitWords.length) % splitWords.length];
```

Comment:
This code needs cleaning up in JS. click the transform to see what I mean

Reply:
Done by manually defining the .js file

Reply:
yeah!

Reply:
To resolve this I think I need to duplicate all the `.ts` into a `.js` file. Before doing that I'll get the technical review.

Reply:
Split this out into a `const idx` line above, and I think the formatting is better now and not worth hand duplicating the `.js`.